docs/writing.rst

```diff
@@ -0,0 +1,95 @@
+.. _writing:
+
+Writing API documentation
+=========================
+
+Documentation for API endpoints starts in the Python files, using docstrings
+and decorators.
+
+
+For an APIView
```

Comment:
Unfortunately the usage of `schema` vs. `schema_for` is not as simple as APIView vs. ViewSet.

APIViews expose the HTTP verbs `get`, `post`, `put`, `patch`, and `delete` as methods, whereas ViewSets expose "abstract" HTTP verbs `list`, `retrieve`, `create`, `update`, `partial_update`, and `destroy` as methods. Any of those methods, if defined by the library user, can be documented directly with `@schema`, and drf-yasg will figure out what's going on.

On either of those base classes, though, you can avoid overriding the HTTP-verb-like methods at all by instead implementing functions like `get_object`, `get_queryset`, `get_serializer`, `perform_create`, etc. This is where `@schema_for` becomes necessary.

I recommend splitting the two subsections along the lines of _Documenting methods directly with @ schema_ and _Documenting methods indirectly with @schema_for_

Reply:
If this contradicts what I've said before, my bad; the more I try to understand DRF the less I feel like I actually understand it.

Reply:
RIGHT! This would go better if I devoted more of my brain to it... I'll make those changes.